File: spec/java/src/io/kaitai/struct/spec/TestCastToImported.java
package io.kaitai.struct.spec;

import io.kaitai.struct.testformats.CastToImported;
import org.testng.annotations.Test;

import static org.testng.Assert.assertEquals;

public class TestCastToImported extends CommonSpec {
    @Test
    public void testCastToImported() throws Exception {
        CastToImported r = CastToImported.fromFile(SRC_DIR + "fixed_struct.bin");

        assertEquals(r.one().one(), 0x50);
        assertEquals(r.oneCasted().one(), 0x50);
    }
}

File: spec/javascript/test_cast_to_imported.js
var assert = require('assert');
var testHelper = require('testHelper');

testHelper('CastToImported', 'src/fixed_struct.bin', function(r) {
  assert.equal(r.one.one, 0x50);
  assert.equal(r.oneCasted.one, 0x50);
});
